git/repo.go

```diff
@@ -345,3 +350,92 @@ func (repo *Repository) ObjectForPath(root Object, pathstr string) (Object, erro
 
 	return node, nil
 }
+
+// ParseCommitList executes a custom git log command of the specified branch of the
+// associated git repository and returns the resulting list of commits as an array.
+func (repo *Repository) ParseCommitList(branch string) ([]wire.CommitListItem, error) {
```

Comment:
Once refactored, this will just return the result of the git log, will refactor name and parameter in the process.